src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,70 @@
+package duke.tasks;
+
+import duke.exceptions.DukeException;
+
+public abstract class Task {
+    private final String description;
+    private boolean isDone;
+
+    public Task(String description) throws DukeException {
+        this(description, false);
+    }
+
+    public Task(String description, boolean isDone) throws DukeException {
+        if (description == null || description.length() == 0) {
+            throw new DukeException("Task description cannot be empty.");
+        }
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Mark the task as completed.
```

Comment:
Suggest to change mark to marks instead 
```suggestion
     * Marks the task as completed.
```